platform/api.htmlui/src/org/netbeans/modules/htmlui/HTMLDialogBase.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.htmlui;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public abstract class HTMLDialogBase {
```

Comment:
The HTMLUI API module no longer requires JavaFX - it can work with different displayers - like the one provided by VSCode. To do that I had to abstract the `HTMLDialogBase` common functionality and provide two implementations - one for Swing (and JavaFX) and one for VSCode.

Reply:
No longer true. The most recent changes in  #3357 use just a single implementation for all `HTMLViewerSpi`s.